src/Surfnet/StepupMiddleware/CommandHandlingBundle/SensitiveData/SensitiveData.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+/**
+ * Copyright 2014 SURFnet bv
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\StepupMiddleware\CommandHandlingBundle\SensitiveData;
+
+use JsonSerializable;
+use Surfnet\Stepup\Exception\InvalidArgumentException;
+use Surfnet\Stepup\Identity\Value\CommonName;
+use Surfnet\Stepup\Identity\Value\DocumentNumber;
+use Surfnet\Stepup\Identity\Value\Email;
+use Surfnet\Stepup\Identity\Value\SecondFactorIdentifier;
+use Surfnet\Stepup\Identity\Value\SecondFactorIdentifierFactory;
+use Surfnet\StepupBundle\Value\SecondFactorType;
+
+class SensitiveData implements JsonSerializable
+{
+    const COMMON_NAME = 'common_name';
+    const EMAIL = 'email';
+    const SECOND_FACTOR_IDENTIFIER = 'second_factor_identifier';
+    const DOCUMENT_NUMBER = 'document_number';
+
+    /**
+     * @var array|null
+     */
+    private $data;
+
+    /**
+     * @param array|null $data A hash of toString-able value objects, indexed by their data keys listed in the constants
+     *    in this class, or null, indicating the sensitive data has been forgotten.
+     * @throws InvalidArgumentException If a data key is not one of the SensitiveData class constants
+     */
+    public function __construct(array $data = null)
+    {
+        if (is_array($data)) {
+            $validKeys = [self::COMMON_NAME, self::EMAIL, self::SECOND_FACTOR_IDENTIFIER, self::DOCUMENT_NUMBER];
+            $invalidKeys = array_diff(array_keys($data), $validKeys);
+            if (count($invalidKeys) > 0) {
+                throw new InvalidArgumentException(
+                    sprintf('Sensitive data contains invalid keys "%s"', join('", "', $invalidKeys))
+                );
+            }
+
+            $this->data = array_map('strval', $data);
```

Comment:
The data is stored as strings internally (as it is constructed with "serialised" data), to be value-objectified when needed. We could also keep the VOs internally, but that would require a factory method that transforms the strings to VOs on instantiation.